viewing_party/party.py

```diff
@@ -1,23 +1,141 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        movie_dict = {'title': title, 'genre': genre, 'rating': rating}
+        return movie_dict
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
 
+def watch_movie(user_data, title):
+    for i in range(len(user_data['watchlist'])):
+        if user_data['watchlist'][i]['title'] == title:
+            user_data['watched'].append(user_data['watchlist'][i])
+            user_data['watchlist'].pop(i)
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
 
+def get_watched_avg_rating(user_data):
+    ratings_total = 0
+    watched_list = user_data["watched"]

```

Comment:
I'm always an advocate for isolating a list into a variable like this! Especially when we have such deeply nested structures!